### PR TITLE
Add premerge/release-qa buildkite pipelines

### PR DESCRIPTION
**Pull request by paulbalaji**
#### Description
- added premerge pipeline
- added release-qa pipeline
- use the `build-worker.sh` that's been moved to `gdk-for-unity-shared-ci`
- drive-by fix of deployment launcher to run `sim_player` deployment without a starting snapshot (as the snapshot was empty anyway)

#### Tests
ran on buildkite and TC

#### Documentation
n/a

#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: ci/bootstrap.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ if [ -d "${SHARED_CI_DIR}" ]; then
     rm -rf "${SHARED_CI_DIR}"
 fi
 
-git clone "${CLONE_URL}" "${SHARED_CI_DIR}"
+git clone "${CLONE_URL}" "${SHARED_CI_DIR}" --branch feature/unify-build-scripts
```
Comment (paulbalaji) [inline ci/bootstrap.sh]: this change will be removed before the final merge

File: ci/build-legacy.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
```
Comment (paulbalaji) [inline ci/build-legacy.sh]: this is identical to the old `ci/build.sh`, and TC will need to be updated to pick up this file instead

File: ci/launch.sh [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+uploadAssembly "${PREFIX}" "${PROJECT_NAME}"
+
+markStartOfBlock "Launching deployments"
+
+dotnet run -p workers/unity/Packages/com.improbable.gdk.deploymentlauncher/.DeploymentLauncher/DeploymentLauncher.csproj -- \
+    create "${PROJECT_NAME}" "${ASSEMBLY_NAME}" \
+    "${ASSEMBLY_NAME}" cloud_launch_large.json snapshots/cloud.snapshot \
+    "${ASSEMBLY_NAME}_sim_players" cloud_launch_large_sim_players.json
+
+if [[ -n "${BUILDKITE-}" ]]; then
```
Comment (jamiebrynes7) [inline ci/launch.sh]: If we never run this on TC its probably safe to remove this.. unless we except users to run this script?

File: ci/build-legacy.sh (hunk 2 above)
Comment (paulbalaji) [reply ci/build-legacy.sh]: yes you are right - I renamed this file back before moving the ci/build.sh to shared-ci

File: ci/launch.sh (hunk 3 above)
Comment (paulbalaji) [reply ci/launch.sh]: Nope. Users can use this as an idea of how to use the deployment launcher through scripts if they want, but we shouldn't expect users to run any of our ci/ scripts

**Comment by paulbalaji** [comment]
> If we are removing the sim players snapshot - what snapshot is the deployment started with now?

The `_sim_player` deployment is started up without a snapshot now, as it was empty before anyway. This works as normal in testing

File: ci/build.sh [hunk 4]
```diff
@@ -10,11 +10,11 @@ ci/bootstrap.sh
```
Comment (jamiebrynes7) [inline ci/build.sh]: File permissions on this got rekt. Should be reverted to `100755`

File: ci/launch.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+markStartOfBlock "Launching deployments"
+
+dotnet run -p workers/unity/Packages/com.improbable.gdk.deploymentlauncher/.DeploymentLauncher/DeploymentLauncher.csproj -- \
+    create "${PROJECT_NAME}" "${ASSEMBLY_NAME}" \
+    "${ASSEMBLY_NAME}" cloud_launch_large.json snapshots/cloud.snapshot \
+    "${ASSEMBLY_NAME}_sim_players" cloud_launch_large_sim_players.json
+
+CONSOLE_URL="https://console.improbable.io/projects/${PROJECT_NAME}/deployments/${ASSEMBLY_NAME}/overview"
+CONSOLE_URL_SIM_PLAYERS="https://console.improbable.io/projects/${PROJECT_NAME}/deployments/${ASSEMBLY_NAME}_sim_players/overview"
```
Comment (jamiebrynes7) [inline ci/launch.sh]: This is way better than the regex approach 😂

File: ci/build.sh (hunk 4 above)
Comment (paulbalaji) [reply ci/build.sh]: good spot, fixed 👍

**Pull request merged**